internal/controller/managedcluster_controller.go

```diff
@@ -420,8 +530,13 @@ func (r *ManagedClusterReconciler) Delete(ctx context.Context, l logr.Logger, ma
 		return ctrl.Result{}, err
 	}
 
+	err = sveltos.DeleteProfile(ctx, r.Client, managedCluster.Namespace, managedCluster.Name)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
Concerning your [suggestion](https://github.com/Mirantis/hmc/pull/362#discussion_r1773348198) to remove Sveltos finalizers. I think it will be better if we don't remove Sveltos finalizer because otherwise the objects being cleaned up in [`reconcileDeleteCommon()`](https://github.com/projectsveltos/addon-controller/blob/29d37c8a90fe47c4518967a4c041a2511fe596c7/controllers/profile_utils.go#L976-L1014) func may be left hanging in the `hmc-system` namespace after the `Profile` object has been deleted.

Reply:
ack. then we should wait for Profile removal before we remove finalizers from the managedcluster object (to not leave hanging resources behind unnoticed)